garrysmod/lua/includes/extensions/net.lua

```diff
@@ -179,3 +148,620 @@ function net.ReadType( typeid )
 
 	error( "net.ReadType: Couldn't read type " .. typeid )
 end
+
+--
+-- Sizes for ints to send
+--
+local TYPE_SIZE = 4
+local UINTV_SIZE = 5
+
+local Char2HexaLookup, Hexa2CharLookup = {}, {}
+
+--
+-- Generate Hexa Lookups
+-- Hexa is a 6-bit English character encoding
+--
+do
+	local HexaRanges = {
+		{ "a", "z" },
+		{ "A", "Z" },
+		{ "0", "9" },
+		{ "_", "_" },
+		{ " ", " " }
+	}
+	
+	local offset = 0
+	
+	for k, v in ipairs( HexaRanges ) do
+	
+		local sChar, eChar = v[ 1 ]:byte( ), v[ 2 ]:byte( )
+		
+		for char = sChar, eChar do
+		
+			local hexa = char - sChar + offset
+			
+			Char2HexaLookup[ char ] = hexa
+			Hexa2CharLookup[ hexa ] = string.char( char )
+		
+		end
+		
+		offset = offset + 1 + eChar - sChar
+	
+	end
+	
+end
+
+--
+-- Converts an ASCII character in to a Hexa Character
+--
+local function CharToHexa( c )
+
+	return Char2HexaLookup[ c ]
+
+end
+
+--
+-- Converts a Hexa character in to an ASCII character
+--
+local function HexaToChar( h )
+
+	return Hexa2CharLookup[ h ]
+	
+end
+
+--
+-- Returns true if the string can be represented in 7-Bit ASCII
+-- Null bytes are not allowed as they are used for termination
+--
+local function Is7BitString( str )
+
+	return str:find( "[ \x80-\xFF%z ]" ) == nil
```

Comment:
Should also bias it so that it checks the first bit instead of the last and then bitshift each character down because commonly used characters are at the end of the ascii table.